src/main/cluster-manager.ts

```diff
@@ -101,6 +100,10 @@ export class ClusterManager extends Singleton {
     }
   }
 
+  protected updateEntityStatus(entity: KubernetesCluster, cluster: Cluster) {
+    entity.status.phase = (cluster.ready && cluster.available && cluster.allowedNamespaces.length > 0) ? "connected" : "disconnected";
```

Comment:
so we display cluster as disconnected when no allowed namespaces present?

Reply:
Yeah, not bullet-proof but currently we mark cluster as connected even it's not connected at all (auth failures etc). If you have better idea to check "connection in usable state" then I'm happy to change this 🙂

Reply:
What is wrong with using `Cluster.accessible`?

Reply:
Because if that is wrong then we should change it instead. Though TBH we should have a lot more derived state (which should be possible once we remove the need to sync Clusters to renderer).

Reply:
> What is wrong with using `Cluster.accessible`?

Nothing, somehow missed that attribute. Now it's using just `cluster.accessible`.